### DISCUSSION TITLE
Feature Request: Improve api design for `VsTable`

### DISCUSSION THREAD

**Discussion by mathiasayivor** [vuesax-alpha/vuesax-alpha#55] [Ideas]
The current structure for `VsTable` works for basic needs, but becomes hard to maintain if there is a need to use at multiple places.

My suggestion is to enable adding data to `VsTable` via props and allow further customisation via slots instead. This would open doors to many other improvements like how sorting is done, etc.

**Comment by tranthinh-coding** [comment]
Currently, the design is for tables with a simple structure. In the future, `TableV2` version may be released with more complex structures to meet everyone's needs. But this will require everyone's support.

**Comment by mathiasayivor** [comment]
I agree with you @tranthinh-coding. This would also help maintain compatibility for users that currently depend on the current design.

I think we should do the same for all components that need a breaking change in API design like #30, etc. Then gradually encourage users to migrate in later releases.
